### DISCUSSION TITLE
i18next with markdown / html

### DISCUSSION THREAD

**Discussion by cjoecker** [sergiodxa/remix-i18next#231] [Ideas]
I would like to add this custom hook to `remix-i18next` to be able to have translations with markdown / HTML. I think it is a common use case. What do you think?

```tsx
import React from "react";
import { useTranslation } from "react-i18next";
import Markdown from "react-markdown";
import remarkGfm from "remark-gfm";

export function useTranslationWithMarkdown() {
	const { t, ...rest } = useTranslation();
	const tMarkdown = React.useCallback(
		(key: string, options?: Record<string, unknown>) => {
			return <Markdown remarkPlugins={[remarkGfm]}>{t(key, options)}</Markdown>;
		},
		[t],
	);
	return { t: tMarkdown, ...rest };
}
```

It can then be used like this
```tsx
const MyComponent = () => {
	const { t } = useTranslationWithMarkdown();
	return (
		<h1>
			{t("yourTranslation")}
		</h1>
	);
};

```

**Comment by sergiodxa** [comment]
Doesn't seem you need remix-i18next for that, as you only need react-i18next for it to work.

**Reply by cjoecker** [reply]
True. But why not to make `remix-i18next` better than `react-i18next` for that? I could submit the issue to `react-i18next` but since they have so many issues, not sure they will prioritize it

**Reply by sergiodxa** [reply]
remix-i18next scope is on remix related functions to integrate Remix (and now RRv7 framework mode) with i18next, specially for server-side code, that's also why it doesn't re-export anything from react-i18next,

**Reply by cjoecker** [reply]
Aa ok, make sense. I will try then some luck on `react-i18next`
